src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
```

Comment:
The checklist advises against using utility classes like `Arrays`. Consider implementing the array copying logic manually or using `System.arraycopy()` as suggested in the checklist.